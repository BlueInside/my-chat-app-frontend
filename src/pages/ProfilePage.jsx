import { useLoaderData } from 'react-router-dom';
import UserProfileHeader from '../components/UserProfileHeader';
import { useAuth } from '../utils/AuthContext';
import UserInfoDisplay from '../components/UserInfoDisplay';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const Container = styled.div`
  display: flex;
  padding: 20px;
  background-color: #f9f9f9;
  border-radius: 8px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  max-width: 600px;
  margin: 0 auto;
  width: 100%;
  flex-direction: column;
  height: 100%;
  justify-content: center;
`;

const EditButton = styled(Link)`
  display: inline-block;
  margin-top: 20px;
  padding: 10px 20px;
  background-color: #007bff;
  color: white;
  text-decoration: none;
  border-radius: 4px;
  width: fit-content;
  &:hover {
    background-color: #0056b3;
  }
`;

function ProfilePage() {
  const user = useLoaderData();
  const { user: currentUser } = useAuth();
  return (
    <Container>
      <UserProfileHeader avatarUrl={user.avatarUrl} username={user.username} />
      <UserInfoDisplay user={user} />
      {currentUser.id === user._id && (
        <EditButton to={'edit'}>Edit Profile</EditButton>
      )}
    </Container>
  );
}

export default ProfilePage;
