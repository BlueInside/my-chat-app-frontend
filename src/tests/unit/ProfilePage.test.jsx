import { render } from '@testing-library/react';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';
import ProfilePage from '../../pages/ProfilePage';

let mockUserData = {
  _id: 'user1',
  username: 'Karol',
  fullName: 'Karol Pulawski',
  bio: 'My very interesting bio',
  avatarUrl: '#',
  dateOfBirth: new Date('1997 05 18'),
};

let router = createMemoryRouter(
  [
    {
      path: '/',
      element: <ProfilePage />,
      loader: () => mockUserData,
    },
  ],
  { initialEntries: ['/'] }
);

describe('ProfilePage Component', async () => {
  it('Renders Profile page properly', () => {
    const { container } = render(<RouterProvider router={router} />);

    expect(container).toMatchSnapshot();
  });
});
