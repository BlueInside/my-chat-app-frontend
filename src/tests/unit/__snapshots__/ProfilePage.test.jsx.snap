// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfilePage Component > Renders Profile page properly 1`] = `
<div>
  <div>
    <div>
      <img
        alt="avatar"
        src="#"
      />
      <p>
        Karol
      </p>
    </div>
    <div>
      <form
        action="/"
        method="get"
      >
        <div>
          <label
            for="fullName"
          >
            Full name
          </label>
          <input
            aria-label="Full name"
            id="fullName"
            name="fullName"
            type="text"
          />
        </div>
        <div>
          <label
            for="avatar"
          >
            Avatar
          </label>
          <input
            aria-label="Avatar"
            id="avatar"
            name="avatar"
            placeholder="Choose a profile picture"
            type="file"
          />
        </div>
        <div>
          <label
            for="dateOfBirth"
          >
            Date of Birth
          </label>
          <input
            aria-label="Date of Birth"
            id="dateOfBirth"
            name="dateOfBirth"
            placeholder="Enter your date of birth"
            type="date"
          />
        </div>
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
